frontend/src/App.tsx

```diff
@@ -40,21 +41,62 @@ import { Emailing } from 'Emailing'
 import { TemplateEditor } from 'TemplateEditor'
 import './App.css'
 import theme from '@core/Constants/Theme'
-import { User, onAuthStateChanged } from 'firebase/auth'
+import { User, onAuthStateChanged, reload } from 'firebase/auth'
 import { AuthProvider, AuthState, PrivateRoute, PublicRoute } from '@auth'
 import { auth } from '@fire'
 import { templatesBucket } from '@fire/firebase'
 import { getDownloadURL, ref } from 'firebase/storage'
 import axios, { AxiosResponse } from 'axios'
-import { CourseProvider, StudentProvider } from '@context'
+import {
+  CourseProvider,
+  StudentProvider,
+  useCourseValue,
+  useStudentValue,
+} from '@context'
 import { TemplateProvider } from '@context/TemplateContext'
+import React from 'react'
 
 const App = () => {
   const [currentUser, setCurrentUser] = useState<User | null>(null)
   const [authState, setAuthState] = useState<AuthState>('loading')
   const axiosAuthInterceptor = useRef<number | null>(null)
   const [networkError, setNetworkError] = useState<string | null>(null)
 
+  const [needsRefresh, setNeedsRefresh] = useState(false)
+  const checkRefreshDuration = 10000 //600000 (10 min)
+
+  const handleClose = (
+    event?: React.SyntheticEvent | Event,
+    reason?: string
+  ) => {
+    if (reason === 'clickaway') {
+      return
+    }
+
+    setNeedsRefresh(false)
+  }
```

Comment:
Is this to close the pop up when you just click on the screen? if so, I don't think you need this.

Reply:
yee, oke